### PR TITLE
Add writability to vk::BufferPointer pointees

### LINKED ISSUE #41: [0010] Address writability of `vk::BufferPointer<T,A>` pointees
**Which proposal does this relate to?**
[0010-vk-buffer-ref.md](proposals/0010-vk-buffer-ref.md)

**Describe the issue or outstanding question.**
The spec mentions `vk::RawBufferStore`, but doesn't explicitly clarify if or how writing through a `vk::BufferPointer<T,A>` is supported. We should clarify this case.

If writing through `vk::BufferPointer<T,A>` is allowed we also need to spec how this works with HLSL's copy-in/copy-out paramter passing as well as pointer address spacing. 

**Additional context**
See [initial review discussion](https://github.com/microsoft/hlsl-specs/pull/37).

### PR DESCRIPTION
**Pull request by greg-lunarg**
Fixes #41

### CODE REVIEW

**Comment by s-perron** [comment]
Could you add an example that shows the buffer being written to?

**Comment by greg-lunarg** [comment]
> Could you add an example that shows the buffer being written to?

@s-perron Done

**Review by s-perron — APPROVED** [review]
LGTM

File: proposals/0010-vk-buffer-ref.md [hunk 1]
```diff
@@ -277,4 +281,35 @@ Here is the SPIR-V for this shader. Note the logical context of the declaration
...
+{
+      Globals_p m_nBufferDeviceAddress;
+};
+
+[[vk::push_constant]] TestPushConstant_t g_PushConstants;
+
+float4 MainPs(void) : SV_Target0
+{
+      float4 vTest = float4(1.0,0.0,0.0,0.0);
+      g_PushConstants.m_nBufferDeviceAddress.Get().g_vTestFloat4 = vTest;
```
Comment (llvm-beanz) [inline proposals/0010-vk-buffer-ref.md:309]: Should we also include the expected SPIR-V operations for this code?

Comment (greg-lunarg) [reply proposals/0010-vk-buffer-ref.md:309]: OK.

Comment (greg-lunarg) [reply proposals/0010-vk-buffer-ref.md:309]: Done.

**Review by llvm-beanz — APPROVED** [review]

**Pull request merged**